moveit_core/robot_trajectory/src/robot_trajectory.cpp

```diff
@@ -504,4 +504,76 @@ bool RobotTrajectory::getStateAtDurationFromStart(const double request_duration,
   return true;
 }
 
+void RobotTrajectory::print(std::ostream& out, std::vector<int> variable_indexes) const
+{
+  size_t num_points = getWayPointCount();
+  if (num_points == 0)
+  {
+    out << "Empty trajectory.";
+    return;
+  }
+
+  std::ios::fmtflags old_settings = out.flags();
+  int old_precision = out.precision();
+  out << std::fixed << std::setprecision(3);
+
+  out << "Trajectory has " << num_points << " points over " << getDuration() << " seconds\n";
+
+  if (variable_indexes.empty())
+  {
+    if (group_)
+    {
+      variable_indexes = group_->getVariableIndexList();
+    }
+    else
+    {
+      size_t variable_count = robot_model_->getVariableCount();
+      variable_indexes.reserve(variable_count);
+      for (size_t i = 0; i < variable_count; i++)
+      {
+        variable_indexes.push_back(i);
+      }
+    }
+  }
+
+  for (size_t p_i = 0; p_i < num_points; p_i++)
+  {
+    const moveit::core::RobotState& point = getWayPoint(p_i);
```

Comment:
We should really finally implement https://github.com/ros-planning/moveit/pull/1162 ...

Reply:
Got it started https://github.com/ros-planning/moveit2/pull/720